Add changelog template for 2025-06-27 (#3598)
neondatabase/website: content/changelog/2025-06-27.md

<a href="https://cursor.com/install-mcp?name=Neon&config=eyJjb21tYW5kIjoibnB4IC15IG1jcC1yZW1vdGVAbGF0ZXN0IGh0dHBzOi8vbWNwLm5lb24udGVjaC9zc2UifQ%3D%3D" style={{display: 'inline-block'}}>
<img src="https://cursor.com/deeplink/mcp-install-dark.svg" alt="Add to Cursor" style={{height: '32px', verticalAlign: 'middle', margin: '0'}} />
</a>

## Enhanced connection security with channel binding

Connection strings and snippets in the Neon Console now include `channel_binding=require` by default, providing stronger protection against man-in-the-middle (MITM) attacks for `psql` and other libpq-based clients:

```shouldWrap
postgresql://alex:AbC123dEf@ep-cool-darkness-a1b2c3d4-pooler.us-east-2.aws.neon.tech/dbname?sslmode=require&channel_binding=require
```

Channel binding works alongside `sslmode=require` to cryptographically link your TLS connection and authentication credentials, making it nearly impossible for attackers to intercept or impersonate your database connections, strengthening security without required client-side root certificate setup.

Most libpq-based clients support this option transparently. For others (e.g., Go's `pgdriver`), compatibility may vary.

> _We recommend updating your connection strings to include `channel_binding=require` if you're using a libpq-based client._

Reviewer: ```suggestion
> We recommend updating your connection strings to include `channel_binding=require` if you're using a libpq-based client.
```

Learn more in our blog post: [Why Postgres needs better connection security defaults](https://neon.com/blog/postgres-needs-better-connection-security-defaults).

## Simplified Neon RLS setup for Neon Auth projects

We've made it easier for you to set up Neon RLS (Row Level Security) for your Neon Auth projects. The Auth page now displays your Stack Auth project details, including the JWKS URL needed for RLS setup.

![Stack Auth project details in Neon Console](/docs/relnotes/neon_auth_jwks.png)

To get started adding RLS to your Neon Auth project:

> 1. Copy the JWKS URL from the **Configuration** tab of your Auth page

Reviewer: ```suggestion
1. Copy the JWKS URL from the **Configuration** tab of your Auth page.
```

[Neon Launchpad](/docs/reference/neon-launchpad) now supports database seeding, allowing developers to automatically populate databases with SQL scripts during database initialization. This feature streamlines the development workflow by enabling instant database setup with sample data. The seeding capability is also available through the Vite plugin integration, making it accessible in Vite-based projects.

To try it from your terminal:

```bash
npx neondb --seed /path/to/file.sql
```

For more details, see:

- [Neondb CLI Changelog](https://github.com/neondatabase/neondb-cli/blob/main/packages/neondb/CHANGELOG.md)
- [Vite Plugin Changelog](https://github.com/neondatabase/neondb-cli/blob/main/packages/vite-plugin-postgres/CHANGELOG.md)

## Scheduled maintenance for Business plans

As announced earlier, we're rolling out scheduled updates that include Postgres version upgrades, security patches, and Neon feature improvements.

These updates are applied during your project's maintenance window or the next time the compute restarts. Most updates take only a few seconds.

> For **Business and Enterprise** plans, updates will begin rolling out on **July 9, 2025**. You can check for update notices and configure your preferred update window — [learn how](/docs/manage/updates#updates-on-paid-plans).

Reviewer: ```suggestion
Updates will begin rolling out on **July 9, 2025** — you'll receive an email notice in advance. You can also check for update notices and configure your preferred update window in the Neon Console — [learn how](/docs/manage/updates#updates-on-paid-plans).
```

> Updates will begin rolling out on **July 9, 2025** — you'll receive an email notice in advance. You can also check for update notices and configure your preferred update window in the Neon Console — [learn how](/docs/manage/updates#updates-on-paid-plans).

Reviewer: ```suggestion
Updates for Business plan projects will begin rolling out on **July 9, 2025** — you'll receive an email notice in advance. You can also check for update notices and configure your preferred update window in the Neon Console — [learn how](/docs/manage/updates#updates-on-paid-plans).
```
